src/test/scala/org/broadinstitute/dsde/firecloud/dataaccess/MockAgoraDAO.scala

```diff
@@ -24,6 +26,44 @@ class MockAgoraDAO extends AgoraDAO {
     Future(List(MockAgoraDAO.agoraPermission))
   }
 
+  override def postMethod(ns: String, name: String, synopsis: String, documentation: String, payload: String)(implicit userInfo: UserInfo): Future[Method] = {
+    // based on what the unit test passed in, return success or failure
+    (ns, name) match {
+      case ("exceptions", "postError") => Future.failed(new AgoraException("postMethod"))
```

Comment:
It's important to use `Future.failed(...)` here instead of directly throwing an exception; otherwise `recover` doesn't catch it in the right way. Think of it this way:

throw Exception: we blew up generating the request, for instance a NPE building the request payload.
Future.failed(Exception): the service we called returned an error response.